sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/DotNetSensor.java

```diff
@@ -94,13 +94,14 @@ private void executeInternal(SensorContext context) {
     }
 
     List<RoslynReport> roslynDirs = reportPathCollector.roslynDirs();
-    if (!roslynDirs.isEmpty()) {
-      Map<String, List<RuleKey>> activeRoslynRulesByPartialRepoKey = activeRoslynRulesByPartialRepoKey(pluginMetadata, context.activeRules()
-        .findAll()
-        .stream()
-        .map(ActiveRule::ruleKey)
-        .collect(toList()));
-      roslynDataImporter.importRoslynReports(roslynDirs, context, activeRoslynRulesByPartialRepoKey, toRealPath);
+    if (roslynDirs.isEmpty()) {
```

Comment:
is it possible that a project without any issues won't end up with any roslynDirs? or are these created during the BEGIN step by the scanner?

Reply:
in `org/sonarsource/dotnet/shared/plugins/AbstractModuleConfiguration.java` I see

```

  /**
   * Returns path of the directory containing all protobuf files, if:
   * - Property with it's location is defined
   * - The directory exists
   * - The directory contains at least one protobuf
   */
  public List<Path> protobufReportPaths() {
```

Reply:
it can be easily tested with an IT that has no issues

Reply:
Generation of the file is configured by `SonarQube.Integration.targets`. It's not configured when `SonarQubeExclude` is present. So in case all projects will have `SonarQubeExclude` we'll fail the analysis. But that kinda makes sense.

Project with no issues will still generate file with empty `results`, I've verified that: 
```
{
  "$schema": "http://json.schemastore.org/sarif-1.0.0",
  "version": "1.0.0",
  "runs": [
    {
      "tool": {
        "name": "Microsoft (R) Visual Basic Compiler",
        "version": "3.5.0.0",
        "fileVersion": "3.5.0-beta4-20153-05 (20b9af91)",
        "semanticVersion": "3.5.0",
        "language": "en-US"
      },
      "results": [
      ]
    }
  ]
}
```

Reply:
This PR is not related to `protobufReportPaths`, see issue description #3593